File: data tratment/tratamento.r
pathName = "C:/Users/mat-v/Desktop/git/Data Science/data tratment"
setwd(pathName)

#Analise Exploratoria
data = read.csv("tempo.csv", sep = ";", na.strings = "", stringsAsFactors = T)
summary(data)

#dados fatoraveis
ct = table(data$Aparencia)
barplot(ct)
ct = table(data$Vento)
barplot(ct)
ct = table(data$Jogar)
barplot(ct)

#dados numericos
summary(data$Temperatura)
boxplot(data$Temperatura)
boxplot(data$Temperatura, outline = F)
summary(data$Umidade)
boxplot(data$Umidade)
boxplot(data$Umidade, outline = F)
data[!complete.cases(data),]

#correção Aparencia
data[data$Aparencia == "menos",]$Aparencia = "sol"
summary(data$Aparencia)
data$Aparencia = factor(data$Aparencia)
summary(data$Aparencia)

#correção temperatura
summary(data$Temperatura)
data[data$Temperatura < -130 | data$Temperatura > 130,]$Temperatura
med = median(data$Temperatura)
data[data$Temperatura < -130 | data$Temperatura > 130,]$Temperatura = med
summary(data$Temperatura)
boxplot(data$Temperatura)

#correção umidade
summary(data$Umidade)
data[data$Umidade<0 | data$Umidade> 100,]$Umidade
med = median(data$Umidade, na.rm = T)
data[is.na(data$Umidade),]$Umidade = med
data[data$Umidade<0 | data$Umidade> 100,]$Umidade = med
summary(data$Umidade)

#correção vento
summary(data$Vento)
data[is.na(data$Vento),]$Vento = "FALSO"
summary(data$Vento)

File: data tratment/tratamento.py
import pandas as pd
import seaborn as srn
import statistics as sts

data = pd.read_csv("tempo.csv", sep=";")
print(data)

agrupado = data.groupby(["Aparencia"]).size()
print(agrupado)

agrupado.plot.bar(color = "gray")

agrupado = data.groupby(["Vento"]).size()
print(agrupado)
agrupado.plot.bar(color="grey")

agrupado = data.groupby(["Jogar"]).size()
agrupado.plot.bar(color="gray")

print(data.isnull().sum())


data.loc[data["Aparencia"] == "menos", "Aparencia"] = "Sol"
agrupado = data.groupby(["Aparencia"]).size()
print(agrupado)


data.loc[(data["Temperatura"] < -130) | (data["Temperatura"] > 130)]
median = sts.median(data["Temperatura"])
data.loc[(data["Temperatura"] < -130) | (data["Temperatura"] > 130), "Temperatura"] = median

print(data["Temperatura"])


data["Umidade"].isnull()
median = sts.median(data["Umidade"])
print(median)
data["Umidade"].fillna(median, inplace=True)

data.loc[(data["Umidade"]< 0) | (data["Umidade"]> 100), "Umidade"] = median
print(data["Umidade"].describe())


data["Vento"].isnull().sum()

agrupamento = data.groupby(["Vento"]).size()
print(agrupamento)

data["Vento"].fillna("FALSO", inplace=True)
print(data["Vento"].isnull().sum())